### DISCUSSION TITLE
Update v6.6 - dev.7

### DISCUSSION THREAD

**Discussion by TBog** [TBog/TBLauncher#212] [Pre-Release]
Change log:
* fix icon cache id (result icon not immediately refreshing after icon changed)
* fix shortcut unpin not removing shortcut immediately from result list
* add `tags menu` action
* add tag list action (same tags as the `tags menu` but can be added to QuickList)
* fix filter rename crash

<hr /><em>This discussion was created from the release <a href='https://github.com/TBog/TBLauncher/releases/tag/v6.6-dev.7'>Update v6.6 - dev.7</a>.</em>

**Comment by TBog** [comment]
Can we make this a stable release? Is there any issue that impacts usability that I'm missing?

**Reply by ltguillaume** [reply]
Well, since dev6 (also in dev7) I'm seeing a really strange thing when tapping on an app in the result list: there's a square around the icon once tapped (sometimes bright white, sometimes dark gray):

![!tblauncher](https://user-images.githubusercontent.com/4777345/137368055-da873723-dc8f-40e3-8b85-5da37eec43eb.gif)

**Reply by TBog** [reply]
Hmm, I don't have this issue, could you please tell me what icon pack you are using and what Android version?

**Reply by ltguillaume** [reply]
I'm on Android 10 (LineageOS 17.1) with no icon pack selected. I'm manually setting the icons, mostly from Whicons, but it doesn't depend on whether the icon is coming from the icon pack or is the default icon with a custom background color (which is #00212121, so transparent). I didn't see this happen before.

**Comment by ltguillaume** [comment]
> add tag list action (same tags as the tags menu but can be added to QuickList)

Fantastic! When you tap a tag from the tag list action, it gets added to the history as if it were an app. Is this the desired behavior? (it's means that every time you use a tag to open an app, both the app and the tag get added to the history...)

**Reply by TBog** [reply]
I'll think about it. Not sure if it's desirable but it's because stuff launched from the result list gets added to the history.

**Reply by ltguillaume** [reply]
Yeah I get that, that's why I asked. It makes sense when reasoning from the "underlying mechanics", but I don't think anyone would like their history being cluttered with tags.

**Reply by ltguillaume** [reply]
Also, you can't remove a tag from history via the context menu. You can only choose `Remove favorite` (even though the tag isn't among the favorites), which does strange things, don't know if it should be there...

**Reply by TBog** [reply]
prevent tags from being added to history 782430a2d9bed35202447696c351c37a93dfc68e

**Reply by ltguillaume** [reply]
Yesss, noice!

**Comment by ltguillaume** [comment]
You can choose `Remove favorite` in the context menu of a tag in the quick list, which 1) doesn't remove it from favorites, because it's not there, 2) doesn't remove it from the quick list either, but 3) _does_ remove the set icon for the tag in question.

**Reply by TBog** [reply]
It does remove it from the QuickList, it just doesn't refresh the QuickList and because of that you still see it there until you restart the launcher.

**Reply by ltguillaume** [reply]
> 
> 
> It does remove it from the QuickList, it just doesn't refresh the QuickList and because of that you still see it there until you restart the launcher.

Ok, that makes sense. But it also removes the icon, though (including from the tags menu and tags list).

**Reply by TBog** [reply]
Yes. I will probably remove the button then.
Everything you customize is added to the favorites.
At this point the favorites do not need to exist anymore. It's just used internally to attribute custom name, icon, position in QuickList for entry items(apps, shortcuts, tags, actions)

It would be best to use a tag instead of the favorites.

It currently is the only way to add an app or shortcut to the QuickList but I can replace the `Add to favorites` button with `Add to QuickList` and it should be ok, right?

**Reply by ltguillaume** [reply]
Yeah I think it would make sense. And perhaps the context menu of a quick list item might then have a `Remove from QuickList` entry?
